tests/common/plugins/conditional_mark/__init__.py

```diff
@@ -420,7 +420,18 @@ def find_all_matches(nodeid, conditions):
 
     for condition in conditions:
         # condition is a dict which has only one item, so we use condition.keys()[0] to get its key.
-        if nodeid.startswith(list(condition.keys())[0]):
+        condition_entry = list(condition.keys())[0]
+        condition_items = condition[condition_entry]
+        if "regex" in condition_items.keys():
+            assert isinstance(condition_items["regex"], bool), \
+                "The value of 'regex' in the mark conditions yaml should be bool type."
+            if condition_items["regex"] is True:
+                match = re.search(condition_entry, nodeid)
```

Comment:
Should we escaped "/" in condition_entry?

Reply:
It's not a must to escape "/" in python:
```
pattern1 = "bfd/test_bfd.py"
pattern2 = r"bfd\/test_bfd.py"
text = "bfd/test_bfd.py"
re.search(pattern1, text)
<re.Match object; span=(0, 15), match='bfd/test_bfd.py'>
re.search(pattern2, text)
<re.Match object; span=(0, 15), match='bfd/test_bfd.py'>
```
And the python re.escape() method doesn't escape “/”:
```
re.escape(pattern1)
'bfd/test_bfd\\.py'
```